File: routes/admin/cateadd.js

var res = require('./lib/resp.js');
var cates = require("./models/cates.js");
var dbUtils = require('./models/dbUtils.js');

if (!session.get("isAdmin")) {
  return res.redirect('/routes/admin/login.js');
}

if (request.getMethod() === 'GET') {
  var allData = cates.find();
  var data = {
    tree: cates.getTree(allData),
  };
  res.render('./views/admin/cateadd.ejs', data);
  return;
}

if (request.getMethod() === 'POST') {
  var data = {
    cat_name: request.formValue('cat_name'),
    intro: request.formValue('cat_desc'),
    parent_id: request.formValue('parent_id'),
    oid: dbUtils.makeOid(),
  };
  cates.add(data);
  res.redirect('./catelist.js');
}

File: routes/admin/cateadd.go
package admin

import (
	"github.com/zengming00/go-testShop/framework"
	"github.com/zengming00/go-testShop/lib"
	"github.com/zengming00/go-testShop/models"
)

func CateAdd(ctx *framework.HandlerContext) {
	var isAdmin = false
	if v, ok := ctx.GetSessionVal("isAdmin"); ok {
		isAdmin = v.(bool)
	}
	if !isAdmin {
		ctx.Redirect("/routes/admin/login.go")
	}

	if ctx.R.Method == "GET" {
		var allData = ctx.Cates.Find()
		var data = map[string]interface{}{
			"tree": ctx.Cates.GetTree(allData),
		}
		ctx.Render("./views/admin/cateadd.html", data, lib.FuncMap)
		return
	}

	if ctx.R.Method == "POST" {
		var data = map[string]interface{}{
			"cat_name":  ctx.R.FormValue("cat_name"),
			"intro":     ctx.R.FormValue("cat_desc"),
			"parent_id": ctx.R.FormValue("parent_id"),
			"oid":       models.MakeOid(),
		}
		ctx.Cates.Add(data)
		ctx.Redirect("./catelist.go")
	}
}
